src/catalog.rs

```diff
@@ -602,6 +609,23 @@ impl PartitionCatalog for DefaultCatalog {
                 _ => Self::to_sqlx_error(e),
             })
     }
+
+    async fn get_orphan_partition_store_ids(&self) -> Result<Vec<String>, Error> {
+        self.repository
+            .get_orphan_partition_store_ids()
+            .await
+            .map_err(Self::to_sqlx_error)
+    }
+
+    async fn delete_partitions(
+        &self,
+        object_storage_ids: Vec<String>,
+    ) -> Result<u64, Error> {
+        self.repository
+            .delete_partitions(object_storage_ids)
+            .await
+            .map_err(Self::to_sqlx_error)
+    }
```

Comment:
Perhaps instead of setting up the boilerplate catalog/interface code, it makes more sense to expose an interface for running ad-hoc queries via sqlx in the `(Default)SeafowlContext` somehow?